lib/checks/report/report.rb

```diff
@@ -0,0 +1,28 @@
+module Checks
+  module Report
+    class Report
```

Comment:
I was actually wondering if we could move the report creation out of the checks. It seems fairly standard but I haven't looked into it yet.

eg would it be sufficient to just get the headers & results from the check object, and have the runner create a report from that (& the whitelist)?